lambdas/aip.py

```diff
@@ -185,42 +193,81 @@ def _get_aip_s3_inventory(self) -> pd.DataFrame:
                 f"S3 Inventory data not found for S3 key: '{self.s3_key}'"
             )
 
-        return inventory_df
+        # index by file key and return
+        return inventory_df.set_index("key")
 
     @staticmethod
     def _decode_base64_sha256(base64_checksum: str | bytes) -> str:
         binary_checksum = base64.b64decode(base64_checksum)
         return binascii.hexlify(binary_checksum).decode("ascii")
 
-    def _get_aip_file_checksums(self) -> dict[str, str]:
+    def _get_aip_file_checksums(
+        self, num_workers: int = CONFIG.checksum_num_workers
+    ) -> dict[str, str]:
         """Retrieve checksums for all files listed in Bagit manifest.
 
         If a SHA256 checksum does not exist, generate one by copying the object onto
-        itself.
+        itself.  This process is performed in parallel via threads by the worker function
+        'process_file_worker' which updates a local dictionary of file-to-checksum.
         """
         if self.manifest_df is None:
             raise ValueError("Bagit manifest data not found")
         if self.s3_inventory is None:
             raise ValueError("S3 Inventory data not found")
 
         file_checksums = {}
-        s3_inventory_indexed = self.s3_inventory.set_index("key")
+        file_checksums_lock = Lock()
```

Comment:
To those not familiar, the `threading.Lock` is a neat thing that allows safe access to an object even with multiple threads running.  We use it on line 239 before updating the dictionary.